### PR TITLE
chore(pkger): add additional documentation for pkger

### PR DESCRIPTION
**Pull request by jsteenb2**

### CODE REVIEW

File: pkger/README.md [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+
+* Translating a declarative [package](#anatomy_of_a_package) file (either JSON | Yaml | Jsonnet) into resources in the platform
+* Exporting existing resources in the form of a pkg (either JSON | Yaml)
+* Managing the state of a pkg's side effects via a stack
+
+## Anatomy of a package
+
+A package is a collection of resource configurations. 
+These resource configurations can be seen in full in the [pkger/testdata](https://github.com/influxdata/influxdb/tree/master/pkger/testdata) directory.
+The package itself does have any state.
```
Comment (sanderson) [inline pkger/README.md]: Should this be "The package itself does *not* have any state."?

File: pkger/README.md [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+* Translating a declarative [package](#anatomy_of_a_package) file (either JSON | Yaml | Jsonnet) into resources in the platform
+* Exporting existing resources in the form of a pkg (either JSON | Yaml)
+* Managing the state of a pkg's side effects via a stack
+
+## Anatomy of a package
+
+A package is a collection of resource configurations. 
+These resource configurations can be seen in full in the [pkger/testdata](https://github.com/influxdata/influxdb/tree/master/pkger/testdata) directory.
+The package itself does have any state. 
+Packages may contain resources that are not uniquely identifiable within the platform and some that are not.
```
Comment (sanderson) [inline pkger/README.md]: Packages may contain resources that are ~not~ uniquely identifiable within the platform and some that are not.

File: pkger/README.md [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+```
+
+The behavior of the servcie includes the following:
+
+1. Dry run a package
+2. Apply a package
+3. Export a package
+4. Initialize a stack
+
+The following sectiosn explore this behavior further.
```
Comment (sanderson) [inline pkger/README.md]: ~sectiosn~ -> sections

File: pkger/README.md [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+
+#### Dry run a package
+
+When a package is submitted for a dry run the service takes the contents of that package and identifies the impact of its application before it is run. This is similar to `terraform plan`. 
+
+> This command is a convenient way to check whether the package matches your expectations without making any changes to real resources. For example, a dry run might be run before committing a change to version control, to create confidence that it will behave as expected.
+
+A dry requires that the package to be dry run has been parsed and graphed. If it has not, the dry run will do so before it attempts the dry run functionality. When a dry run is executed, the caller will have returned a summary of the package and a detailed diff of the impact of the package were it to be applied.
```
Comment (sanderson) [inline pkger/README.md]: A dry **run** requires the package to be parsed and graphed.

File: pkger/README.md [hunk 5]
```diff
@@ -0,0 +1,639 @@
...
+
+
+#### Apply a package
+
+When a package is submitted to be applied, the service takes the contents of that package and identifies the impact of its application before it is run (Dry Run). It then brings the platform to the desired state of the package.
+
+> Apply is used to apply the changes required to reach the desired state of the package
+
+If a package had not been verified by a drun run when applying, it will be done to identify existing state within the platform. This existing state has to be maintained to account for an unexpected event that stops the package from being applied. The side effects created from the application will all be rolled back at this point. If a resource was newly created during the application, it will be removed. For a resource that existed in the platform, it will be returned to its state from before the application took place. The guarantee of state consistency is a best attempt. It is not bullet proof. However, a user can reapply the package and arrive at their desired state therafter. Upon successful application of a package, a summary will be provided to the user.
```
Comment (sanderson) [inline pkger/README.md]: ~drun~ -> dry

**Comment by jsteenb2** [comment]
awesome feedback @sanderson, I went ahead and updated the doc per your feedback 👍

**Pull request merged**